node/node_handler.go

```diff
@@ -84,9 +84,11 @@ func (node *Node) HandleNodeMessage(
 						if block.ShardID() == 0 {
 							utils.Logger().Info().
 								Msgf("Beacon block being handled by block channel: %d", block.NumberU64())
-							go func(blk *types.Block) {
-								node.BeaconBlockChannel <- blk
-							}(block)
+							if block.IsLastBlockInEpoch() {
+								go func(blk *types.Block) {
+									node.BeaconBlockChannel <- blk
```

Comment:
Note to also change the beacon chain block broadcasting logic to only broadcast epoch block.